cli/src/complete.rs

```diff
@@ -106,10 +122,23 @@ fn get_jj_command() -> Result<std::process::Command, CommandError> {
         .disable_version_flag(true)
         .disable_help_flag(true)
         .ignore_errors(true)
+        // Here, allow_external_subcommands fixes a weird issue. Without it,
+        // parsing GlobalArgs will fail with the message that a required arg
+        // is missing, where the required arg is a boolean flag. This seems
+        // unexpected, because missing boolean flags are usually treated as
+        // false. It is also not clear to me why allow_external_subcommands
+        // changes this behavior. See the discussion in the clap repo:
+        // https://github.com/clap-rs/clap/discussions/5812
+        .allow_external_subcommands(true)
```

Comment:
There's an incoming patch for this in clap:
https://github.com/clap-rs/clap/pull/5813

Let's wait with this PR until we can get rid of this workaround.

Reply:
The fix in clap is merged and released.